docs/guide/migration/upgrading.md

```diff
@@ -0,0 +1,134 @@
+
+### Content Upgrader
+
+While completely manual code migration is possible, the *Content* Upgrader tool is basically essential for data conversion. Writing custom data migrations would not be able to do many thing differently from the official tool and there is very little to gain by doing so.
+
+The Content Upgrader tool does not change anything on the original Apostrophe 2 database. Instead, it reads that original database and creates a *new database* using a name the developer provides. That new database will contain the original data, converted for use in Apostrophe 3.
+
+Developers will install the Content Upgrader as a module within the A2 project. This allows it to access schemas and other important project information. **See the [Content Upgrader README](https://www.npmjs.com/package/@apostrophecms/content-upgrader) for full instructions.**
+
+#### Limitations
+
+There are a few limitations in the Content Upgrader to understand before using it.
+
+- Users and groups are not migrated. This is because the user roles of A3 differ in design from the permissions groups of A2. Copying users directly could create security issues. Create new accounts on the A3 project or arrive at your own migration strategy.
+- A2 has a built-in `apostrophe-images` widget type that acts as a slideshow when multiple images are selected. A3 only has a built-in single-image `@apostrophecms/image` widget type. By default `apostrophe-images` will be upgraded to `@apostrophecms/image`, with only the first image present in each. However you can use the `mapWidgetTypes` option (see the README) to override this mapping during the upgrade.
+- Since A3 does not have a built-in cropping feature yet, there is currently no accommodation for it in A3's `@apostrophecms/image` widget type. The tool attempts to carry over the cropping data in the format which is expected to work in A3 in the near future.
+
+### Code Upgrader
+
+The *Code* Upgrader tool support upgrading codebases for **full Apostrophe projects** (websites) and **installable modules**. It has two major roles in converting an A2 codebase to use Apostrophe 3:
+
+1. It will lint an Apostrophe codebase for A2 syntax and structure that must change. This is its `lint` command.
+2. It will *make* many of those code changes for you. This is its `upgrade` command.
+
+Additionally, there is a `reset` command that can undo the automated changes. Always use a new git branch during this process as well to have an additional way to roll back changes.
+
+One reason we recommend using this tool to execute changes on a full project is that it will make minimal code adjustments for A3 use. This is important because additional changes (such as complete module name changes), could unnecessarily break compatibility with the [upgraded database](#content-upgrader). After running the automatic code upgrade, take care in the final code changes to avoid effecting data compatibility.
```

Comment:
We're probably overstating the benefits here, and it's also not clear why the developer would make "complete module name changes."

effecting -> affecting

Reply:
I'll update the example to changing the field name instead. Less radical, still breaks complete compatibility.